static/js/main.js

```diff
@@ -2,20 +2,18 @@ function initializeThemeSwitcher() {
     const themeSwitch = document.querySelector('.theme-switcher input[type="checkbox"]');
 
     themeSwitch.addEventListener("change", function(e) {
+        let theme = 'light';
         if (e.target.checked) {
-            document.body.className = "dark-theme";
-            localStorage.setItem("theme", "dark");
-        } else {
-            document.body.className = "light-theme";
-            localStorage.setItem("theme", "light");
+            theme = 'dark';
         }
+
+        document.documentElement.setAttribute('theme', theme);
+        localStorage.setItem('theme', theme);
     }, false);
 
-    const isDarkOS = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
-    const theme = localStorage.getItem("theme");
-    if (theme === "dark" || isDarkOS) {
-        document.body.className = "dark-theme";
-        themeSwitch.checked = true;
+    const theme = localStorage.getItem('theme');
+    if (theme !== undefined) {
```

Comment:
`localStorage.getItem('nonexistent')` seems to return `null` and not undefined. Might want to either change to `!=` or to change `undefined` -> `null`

Reply:
@sudodoki thanks for spotting this. Initially, I used a non-strict comparison which worked just fine for me but decided that I should clean it before submitting a pull request. I guess this was a bad idea.

I force pushed to keep vas3k's repo clean in case he decides to merge the request.